### PR TITLE
Add test for ximgproc/edgeboxes and fix up Python sample to work with returning scores

### PR DESCRIPTION
**Pull request by saskatchewancatch**
### This pullrequest changes

Adds a regression test for the edgeboxes implementation in ximgproc, and fixes up a Python sample for the edgeboxes which now works for returning scores.

### CODE REVIEW

**Comment by saskatchewancatch** [comment]
Hopefully that takes care of the VS compiler warning. Feedback on this test case appreciated. The test image of the bird is a simple proposal case with an obvious answer and this test should guard against most regressions.

**Comment by asmorkalov** [comment]
:+1:

**Comment by asmorkalov** [comment]
@saskatchewancatch Unfortunately CI bot does not run tests for Contrib modules, because they are not stable enough. I run the test manually, and see failure:

**Comment by alalek** [comment]
> CI bot does not run tests for Contrib modules

This test has been launched.

[OpenCV buildbot](http://pullrequest.opencv.org/buildbot/builders/precommit-contrib_linux64/builds/13162):
```
[----------] 1 test from ximpgroc_Edgeboxes
[ RUN      ] ximpgroc_Edgeboxes.regression
[       OK ] ximpgroc_Edgeboxes.regression (2163 ms)
[----------] 1 test from ximpgroc_Edgeboxes (2163 ms total)
```

Additional `opencv_extra` patch should be applied: https://github.com/opencv/opencv_extra/pull/628
BTW, It is better to store testdata in lossless formats, like PNG. JPEG read results may vary between platforms / versions / compiler flags.

Also test message should be improved for case of missing data:
```.diff
 Mat testImg = imread(testImagePath);
+ASSERT_FALSE(testImg.empty()) << "Can't read test image: " << testImagePath;
```

+ similar message for empty/missing model.

**Comment by asmorkalov** [comment]
I re-done on another host with Ubuntu 16.04 and everything passed. So we can go ahead with the path.

File: modules/ximgproc/test/test_edgeboxes.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+#include "test_precomp.hpp"
+
+namespace opencv_test { namespace {
+
+TEST(ximpgroc_Edgeboxes, regression)
+{
+	//Testing Edgeboxes implementation by asking for one proposal
+	//on a simple test image from the PASCAL VOC 2012 dataset.
```
Comment (alalek) [inline modules/ximgproc/test/test_edgeboxes.cpp]: tabs => 4 spaces

File: modules/ximgproc/test/test_edgeboxes.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    edgeboxes->setMaxBoxes(1);
+    edgeboxes->getBoundingBoxes(edgeImage, edgeOrientations, boxes, scores);
+
+    //We asked for one proposal and thus one score, we better get one back only.
+    ASSERT_TRUE(boxes.size() == 1);
+    ASSERT_TRUE(scores.size() == 1);
+
+    //Check the proposal and its score.
+    EXPECT_FLOAT_EQ(scores[0], expectedScore);
+    EXPECT_TRUE(expectedProposal.x == boxes[0].x && expectedProposal.y == boxes[0].y && expectedProposal.height == boxes[0].height && expectedProposal.width == boxes[0].width);
```
Comment (alalek) [inline modules/ximgproc/test/test_edgeboxes.cpp]: Please split on multiple checks.
`==` checks should be replaced by `EXPECT_EQ(extected_value, actual_value)` (for integers) or `EXPECT_NEAR(expected, actual, abs_err)`

The same note is about `ASSERT_TRUE(... == ...)` above.

**Comment by saskatchewancatch** [comment]
> > CI bot does not run tests for Contrib modules
> 
> This test has been launched.
> 
> [OpenCV buildbot](http://pullrequest.opencv.org/buildbot/builders/precommit-contrib_linux64/builds/13162):
> 
> ```
> [----------] 1 test from ximpgroc_Edgeboxes
> [ RUN      ] ximpgroc_Edgeboxes.regression
> [       OK ] ximpgroc_Edgeboxes.regression (2163 ms)
> [----------] 1 test from ximpgroc_Edgeboxes (2163 ms total)
> ```
> 
> Additional `opencv_extra` patch should be applied: [opencv/opencv_extra#628](https://github.com/opencv/opencv_extra/pull/628)
> BTW, It is better to store testdata in lossless formats, like PNG. JPEG read results may vary between platforms / versions / compiler flags.
> 
> Also test message should be improved for case of missing data:
> 
> ```diff
>  Mat testImg = imread(testImagePath);
> +ASSERT_FALSE(testImg.empty()) << "Can't read test image: " << testImagePath;
> ```
> 
> * similar message for empty/missing model.

Loading image and errorring out with a nice message seems like a common thing in the tests. Perhaps at some point testing infra could be enhanced with a utility function that loads+fails on not found?

**Comment by asmorkalov** [comment]
@alalek LGTM :+1: .
@saskatchewancatch Please squash commits.

**Pull request merged**